src/pvrclient-argustv.cpp

```diff
@@ -1492,20 +1500,34 @@ bool cPVRClientArgusTV::FindRecEntryUNC(const std::string &recId, std::string &r
   if (iter == m_RecordingsMap.end())
     return false;
 
-  recEntryURL = ToUNC(iter->second.c_str());
+  recEntryURL = ToUNC(iter->second);
   if (recEntryURL == "")
     return false;
 
   return true;
 }
 
+bool cPVRClientArgusTV::FindRecEntry(const std::string &recId, std::string &recEntryURL)
+{
+  auto iter = m_RecordingsMap.find(recId);
+  if (iter == m_RecordingsMap.end())
+    return false;
+
+  recEntryURL = iter->second;
+  if (!g_szUser.empty())
```

Comment:
Instead of doing this check every time before calling `InsertUser`, you could do it at the beginning of `InsertUser`.